### PR TITLE
Adding in the remaining unit tests for constraints, filled out some TODOs

### PR DESCRIPTION
**Pull request by tbarnes94**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)
Feature # (details)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Is it a  breaking change which will impact consuming tool(s).

- TestValidateDeployment_UniqueServiceSelectors [x]
- TestValidateDeployment_RestrictedTaints [x]
- TestValidateDeployment_PodEnforceAntiaffinity [x]
- TestValidateDeployment_DisallowedBadPodDisruptionBudget [x]
- TestValidateDeployment_All [x]

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/safeguards/tests/restricted-taints/RT-success-manifest.yaml [hunk 1]
```diff
@@ -1,21 +1,13 @@
-apiVersion: apps/v1
-kind: Deployment
+apiVersion: v1
+kind: Node
 metadata:
-  name: my-deployment
+  name: my-error-node
```
Comment (NickKeller) [inline pkg/safeguards/tests/restricted-taints/RT-success-manifest.yaml:4]: should be success?

File: pkg/safeguards/safeguards.go [hunk 2]
```diff
@@ -36,37 +36,37 @@ func init() {
 }
 
 // GetManifestViolations takes in a list of manifest files and returns a slice of ManifestViolation structs
-func GetManifestViolations(ctx context.Context, manifestFiles []ManifestFile) ([]ManifestViolation, error) {
+func GetManifestViolations(ctx context.Context, manifestFiles []ManifestFile) ([]ManifestResult, error) {
```
Comment (NickKeller) [inline pkg/safeguards/safeguards.go:39]: rename function to GetManifestResults?

Comment (tbarnes94) [reply pkg/safeguards/safeguards.go]: done

File: pkg/safeguards/tests/restricted-taints/RT-success-manifest.yaml (hunk 1 above)
Comment (tbarnes94) [reply pkg/safeguards/tests/restricted-taints/RT-success-manifest.yaml]: done

**Review by NickKeller — APPROVED** [review]

**Review by davidgamero — APPROVED** [review]
lgtm can we document that is merges main's changes and isn't just a huge new set of changes?

**Pull request merged**